aten/src/ATen/core/Generator.cpp

```diff
@@ -0,0 +1,316 @@
+#include "ATen/core/Generator.h"
+
+namespace at {
+
+/*
+* GeneratorState copy and assign constructor implementations
+*/
+GeneratorState::GeneratorState(const GeneratorState& other) {
+  this->device = other.device;
+  this->current_seed = other.current_seed;
+  this->device_type = other.device_type;
+  this->philox_offset_per_thread.exchange(other.philox_offset_per_thread);
+  new (&(this->cpu_engine)) std::mt19937_64(other.cpu_engine);
```

Comment:
Why are you working so hard? At least on GCC and Clang, I seem to have copy assignment on this type: https://godbolt.org/z/uiHS3A

Reply:
You are right. I am myself asking now why. I think what happened was I was copying a lot of the initialization style from the old TH, like THCRandom_init etc. and was going through some indecision about making mt19937 a pointer or variable and other shenanigans. I will fix this and use the copy constructor of mt19937.

Reply:
Fixed this by using mt19937 copy constructor.